test/src/test/java/org/corfudb/runtime/view/AbstractViewTest.java

```diff
@@ -524,8 +524,13 @@ protected void startCompaction(CorfuRuntime rt, LogUnitServer logUnitServer) {
         // stop periodical tasks to prevent race condition
         rt.getGarbageInformer().stop();
 
+        // wait until all garbage decisions are sent to logUnit servers.
+        rt.getGarbageInformer().waitUntilAllTasksFinish();
+
         // send garbage decisions to logUnit servers
-        rt.getGarbageInformer().gcUnsafe();
+        while (rt.getGarbageInformer().getGarbageReceivingQueue().size() > 0) {
+            rt.getGarbageInformer().gcUnsafe();
+        }
```

Comment:
Put this logic into GarbageInformer together with waitUntilAllTasksFinish  as we discussed.

Reply:
done.